### ISSUE TITLE
Obsolete service containers stuck in "stopping" state and cannot be removed.

### ISSUE THREAD

**Issue by yeta-io** [rancher/rancher#7102]
**Rancher Version:** 1.2.1, also reproduced on 1.2.0

**Docker Version:** Docker version 1.12.3, build 6b644ec

**OS and where are the hosts located? (cloud, bare metal, etc):** vps, ubuntu zesty

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)** single rancher + slaves for services

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)** cattle

**Steps to Reproduce:** create service stack, add service, do few times service application upgrades.

**Results:** previous (old versions) of service instance containers displayed in "stopping" state, the containers can't be removed. Also I see these containers on hosts dashboard as "in progress".
Tried to restart rancher or agent - no luck.

**Expected:** instances should be removed and purged, or at least delete action should remove them.

![rancher 2016-12-17 03-21-26](https://cloud.githubusercontent.com/assets/23072552/21283252/258db5f6-c408-11e6-9cbb-86c29071818c.png)

**Comment by yeta-io** [comment]
I see in rancher server logs a lot of messages like, may be it anyhow related to the issue?

```
* 'Instance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string'
* 'Instance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string'
* 'Instance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string'
* 'Instance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string'
* 'Instance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string'
* 'Instance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string'
* 'Instance.Data.DockerInspect.State.Health.Log[3].End' expected a map, got 'string'
* 'Instance.Data.DockerInspect.State.Health.Log[3].Start' expected a map, got 'string'
* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string'
* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string'
* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string'
* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string'
* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string'
* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string'
* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[3].End' expected a map, got 'string'
* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[3].Start' expected a map, got 'string'
2016-12-17 04:32:48,048 ERROR [f6bc4b98-c373-43c0-b821-3483a83870c7:6385] [instance:223->instanceHostMap:80] [instance.stop->(InstanceStop)->instancehostmap.deactivate] [] [ecutorService-9] [c.p.e.p.i.DefaultProcessInstanceImpl] Agent error for [compute.instance.deactivate.reply;agent=15]: 8 error(s) decoding:
```

**Comment by knepperjm** [comment]
I'm seeing this too.  Failed health check causes non-stop rescheduling of old containers even after deleting / purging of old stack and stopped containers in UI with a re-deploy of a new stack with updated docker image / etc.

**Comment by knepperjm** [comment]
`2/2017 1:42:02 PMtime="2017-02-02T21:42:02Z" level=error msg="Error processing event" err="10 error(s) decoding:\n\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[3].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[3].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[4].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[4].Start' expected a map, got 'string'" eventId=33e7a01e-b5a7-4916-a906-a1834e707097 eventName=storage.volume.remove resourceId=1vspm118552
2/2/2017 1:42:02 PMtime="2017-02-02T21:42:02Z" level=error msg="Error processing event" err="10 error(s) decoding:\n\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[3].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[3].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[4].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[4].Start' expected a map, got 'string'" eventId=16b5d979-0eb0-40c1-84bb-cdf8d8f01505 eventName=storage.volume.remove resourceId=1vspm118498
2/2/2017 1:42:02 PMtime="2017-02-02T21:42:02Z" level=info msg="Received event: Name: storage.volume.remove, Event Id: fb6e746a-fb41-4ad1-ad86-b21db43b1914, Resource Id: 1vspm118545"
2/2/2017 1:42:02 PMtime="2017-02-02T21:42:02Z" level=error msg="Error processing event" err="10 error(s) decoding:\n\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[3].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[3].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[4].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[4].Start' expected a map, got 'string'" eventId=fb6e746a-fb41-4ad1-ad86-b21db43b1914 eventName=storage.volume.remove resourceId=1vspm118545
2/2/2017 1:42:02 PMtime="2017-02-02T21:42:02Z" level=info msg="Received event: Name: storage.volume.remove, Event Id: a4c99914-a82a-4929-93be-6367690c3471, Resource Id: 1vspm118544"
2/2/2017 1:42:02 PMtime="2017-02-02T21:42:02Z" level=error msg="Error processing event" err="10 error(s) decoding:\n\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[3].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[3].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[4].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[4].Start' expected a map, got 'string'" eventId=a4c99914-a82a-4929-93be-6367690c3471 eventName=storage.volume.remove resourceId=1vspm118544
2/2/2017 1:42:02 PMtime="2017-02-02T21:42:02Z" level=info msg="Received event: Name: storage.volume.remove, Event Id: 150150e7-25a8-47d0-abee-f52f0bf194eb, Resource Id: 1vspm118630"
2/2/2017 1:42:02 PMtime="2017-02-02T21:42:02Z" level=error msg="Error processing event" err="10 error(s) decoding:\n\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[3].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[3].Start' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[4].End' expected a map, got 'string'\n* 'Volume.Instance.Data.DockerInspect.State.Health.Log[4].Start' expected a map, got 'string'" eventId=150150e7-25a8-47d0-abee-f52f0bf194eb eventName=storage.volume.remove resourceId=1vspm118630
2/2/2017 1:42:02 PMtime="2017-02-02T21:42:02Z" level=info msg="Received event: Name: compute.instance.remove, Event Id: f289d2f3-48cd-4f28-b580-ff841b435f63, Resource Id: 1ihm114364"
2/2/2017 1:42:02 PMtime="2017-02-02T21:42:02Z" level=error msg="Error processing event" err="20 error(s) decoding:\n\n* 'Instance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string'\n* 'Instance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string'\n* 'Instance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string'\n* 'Instance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string'\n* 'Instance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string'\n* 'Instance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string'\n* 'Instance.Data.DockerInspect.State.Health.Log[3].End' expected a map, got 'string'\n* 'Instance.Data.DockerInspect.State.Health.Log[3].Start' expected a map, got 'string'\n* 'Instance.Data.DockerInspect.State.Health.Log[4].End' expected a map, got 'string'\n* 'Instance.Data.DockerInspect.State.Health.Log[4].Start' expected a map, got 'string'\n* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string'\n* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string'\n* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string'\n* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string'\n* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string'\n* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string'\n* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[3].End' expected a map, got 'string'\n* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[3].Start' expected a map, got 'string'\n* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[4].End' expected a map, got 'string'\n* 'Instance.Volumes[0].Instance.Data.DockerInspect.State.Health.Log[4].Start' expected a map, got 'string'" eventId=f289d2f3-48cd-4f28-b580-ff841b435f63 eventName=compute.instance.remove resourceId=1ihm114364`

**Comment by vincent99** [comment]
This was fixed in https://github.com/rancher/agent/commit/4c22fc5952d780f76f1363934bbf8998758f8d4c and shipped with agent `v0.6.1` (Rancher `v1.3.0`).  As a workaround on 1.2.x you can find the container which has a docker healthcheck defined and remove it.

**Comment by em0ney** [comment]
I have been experiencing this same behaviour, but my logs look different.  Same steps to reproduce as reported but I'm running HA.

```
Error processing event" err="Failed to stop container b56271a2-7b8f-4e3d-b89c-f2188b901cd4" eventId=afc83c2c-cfbe-4f1d-a0be-3e6fce06b07a eventName=compute.instance.deactivate resourceId=1ihm121
8/10/2017 2:56:13 PMtime="2017-08-10T04:56:13Z" level=info msg="Received event: Name: compute.instance.deactivate, Event Id: e761d3f7-a823-4bb1-a103-8be1304ff2c8, Resource Id: 1ihm39"
8/10/2017 2:56:13 PMtime="2017-08-10T04:56:13Z" level=info msg="Received event: Name: compute.instance.deactivate, Event Id: c980aa06-22cb-4773-b887-e9e4421df63e, Resource Id: 1ihm111"
8/10/2017 2:56:13 PMtime="2017-08-10T04:56:13Z" level=info msg="Received event: Name: compute.instance.deactivate, Event Id: 1908bdf5-1047-4484-94e4-02d8485ec280, Resource Id: 1ihm118"
8/10/2017 2:56:27 PMtime="2017-08-10T04:56:27Z" level=error msg="Error processing event" err="Failed to stop container b78db52c-fd0d-448b-9774-37ac0b57d1b0" eventId=a11e0d57-190d-413c-90e5-546ce4dc3eb7 eventName=compute.instance.deactivate resourceId=1ihm34
8/10/2017 2:56:27 PMtime="2017-08-10T04:56:27Z" level=info msg="Received event: Name: compute.instance.deactivate, Event Id: e96494f7-004b-4a11-850c-2fa7175522fe, Resource Id: 1ihm34"
8/10/2017 2:56:27 PMtime="2017-08-10T04:56:27Z" level=error msg="Error processing event" err="Failed to stop container d5f319b4-b1f6-4b0c-ab2c-9735ec196f10" eventId=cfab9281-1049-42e3-a739-6c15aa3715a8 eventName=compute.instance.deactivate resourceId=1ihm36
8/10/2017 2:56:27 PMtime="2017-08-10T04:56:27Z" level=error msg="Error processing event" err="Failed to stop container 6cc50dcb-eecc-4f2a-af20-7888417f2c6c" eventId=9b0313ae-4983-4ad5-a68b-f2c43e74ac1a eventName=compute.instance.deactivate resourceId=1ihm112
8/10/2017 2:56:27 PMtime="2017-08-10T04:56:27Z" level=info msg="Received event: Name: compute.instance.deactivate, Event Id: c4adee7d-7a66-437d-a288-1bc384440561, Resource Id: 1ihm36"
8/10/2017 2:56:27 PMtime="2017-08-10T04:56:27Z" level=error msg="Error processing event" err="Failed to stop container 4fdf8739-ca99-433e-8108-67e6c2e38c52" eventId=166465f1-bbc9-45b2-aa75-9c6a9519ccf5 eventName=compute.instance.deactivate resourceId=1ihm35
8/10/2017 2:56:27 PMtime="2017-08-10T04:56:27Z" level=info msg="Received event: Name: compute.instance.deactivate, Event Id: 9550e57b-218a-4fed-b18b-47ca25511977, Resource Id: 1ihm119"
8/10/2017 2:56:27 PMtime="2017-08-10T04:56:27Z" level=info msg="Received event: Name: compute.instance.deactivate, Event Id: f38761cd-6390-4c5f-bfab-1629c7e77af9, Resource Id: 1ihm122"
8/10/2017 2:56:27 PMtime="2017-08-10T04:56:27Z" level=info msg="Received event: Name: compute.instance.deactivate, Event Id: e9b4847e-f1d4-4ea4-8257-5e72721eb196, Resource Id: 1ihm121"
```
